From 49a2b55e0e1e265c1393c04c9df8e73f30dccd33 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston <jeremyhu@apple.com>
Date: Sun, 28 Mar 2010 13:46:23 -0700
Subject: [PATCH 10/14] XQuartz: Re-query dixScreenOrigins as the value could've changed.

Fix a regression in 9c9c3a85b094a3c7b2763a572715d710325091aa

Signed-off-by: Jeremy Huddleston <jeremyhu@apple.com>
Acked-by: Adam Jackson <ajax@redhat.com>
(cherry picked from commit 28a5f14b4089dccb8045cc4fdc923542a73dd22d)
---
 hw/xquartz/quartz.c |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git xorg-server-1.7.6/hw/xquartz/quartz.c xorg-server-1.7.6/hw/xquartz/quartz.c
index 59107be..a611854 100644
--- xorg-server-1.7.6/hw/xquartz/quartz.c
+++ xorg-server-1.7.6/hw/xquartz/quartz.c
@@ -268,8 +268,11 @@ void QuartzUpdateScreens(void) {
     DarwinAdjustScreenOrigins(&screenInfo);
     quartzProcs->UpdateScreen(pScreen);
     
-    sx = x + darwinMainScreenX;
-    sy = y + darwinMainScreenY;
+    /* DarwinAdjustScreenOrigins or UpdateScreen may change dixScreenOrigins,
+     * so use it rather than x/y
+     */
+    sx = dixScreenOrigins[pScreen->myNum].x + darwinMainScreenX;
+    sy = dixScreenOrigins[pScreen->myNum].y + darwinMainScreenY;
     
     /* Adjust the root window. */
     pRoot = WindowTable[pScreen->myNum];
-- 
1.7.0.2

